### PR TITLE
feat(core): Implement a pebble function fileSize() that returns the file size in bytes

### LINKED ISSUE #5902: Add a pebble function `fileSize()` that returns the file size in bytes
### Feature description

This is needed for use cases when you need to check the file size within a condition and perhaps skip some downstream tasks if the file is empty.

This can be a function that takes an internal storage file URI as a single input parameter: `fileSize(URI)`

### PR DESCRIPTION
**Pull request by benjoEK1337**
### What changes are being made and why?

---

The new pebble function is created to return the file size that takes the internal storage URI. The implementation followed some logic from the `read()` pebble function regarding the way it gets the files and handles the cases around current and parent execution. The only difference is that the **fileSize** function is implemented to handle **only** the case when the path is Internal Storage URI, meaning it starts with "kestra:///".  The namespace file is not covered.

### How the changes have been QAed?

---
I tested this feature through unit tests. I am open to suggestions on how to test it further.

### CODE REVIEW

File: core/src/main/java/io/kestra/core/runners/pebble/functions/FileSizeFunction.java
Comment (benjoEK1337) [inline core/src/main/java/io/kestra/core/runners/pebble/functions/FileSizeFunction.java:19]: There is a lot of sharing logic between the **read** pebble function and this function, so the logic could be centralized into the new class but didn't want to overengineer it, since it's my first issue here.

Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/runners/pebble/functions/FileSizeFunction.java:19]: Yes, an abstract FileFunction could be a good idea.
But let's keep that for later, I'm OK to merge it as is.

**Review by loicmathieu — APPROVED** [review]
LGTM, thanks

**Comment by benjoEK1337** [comment]
Is there something else related to the task I need to do after this is merged? 😄

**Comment by MilosPaunovic** [comment]
Just to wait until the CI passes.

**Pull request merged**
